File: math/pow/pow.c


double myPow(double x, long long n)
{
	if(!n) return 1;
	if(n <0)
	{
		n = -n;
		x = 1 /x;
	}
	return n &1 ? myPow(x,n -1) *x : myPow(x*x,n >>1);
}

File: math/pow/pow.java


// LeetCode Q50. Pow(x, n)
// Implement pow(x, n), which calculates x raised to the power n (i.e., xn).


public double myPow(double x, int n) {

	double r = 1;
	if(n <0) {

		n = -n;
		x = 1 /x;
	}
	while(n != 0) {
		if((n &1) != 0) {
			r *= x;
		}
		x *= x;
		n >>>= 1;
	}
	return r;
}

